### PR TITLE
fix(elb): be able to remove a description

### PR DESCRIPTION
**Pull request by ShiChangkuo**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
$ make testacc TEST=./huaweicloud/services/acceptance/lb TESTARGS='-run=TestAccLBV2LoadBalancer_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/lb -v -run=TestAccLBV2LoadBalancer_basic -timeout 360m -parallel 4
=== RUN   TestAccLBV2LoadBalancer_basic
=== PAUSE TestAccLBV2LoadBalancer_basic
=== CONT  TestAccLBV2LoadBalancer_basic
--- PASS: TestAccLBV2LoadBalancer_basic (129.63s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/lb        129.703s

$ make testacc TEST=./huaweicloud/services/acceptance/lb TESTARGS='-run=TestAccLBV2Listener_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/lb -v -run=TestAccLBV2Listener_basic -timeout 360m -parallel 4
=== RUN   TestAccLBV2Listener_basic
=== PAUSE TestAccLBV2Listener_basic
=== CONT  TestAccLBV2Listener_basic
--- PASS: TestAccLBV2Listener_basic (144.49s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/lb        144.549s
```

### CODE REVIEW

File: huaweicloud/services/elb/resource_huaweicloud_elb_listener.go [hunk 1]
```diff
@@ -320,7 +320,8 @@ func resourceListenerV3Update(ctx context.Context, d *schema.ResourceData, meta
 			updateOpts.Name = d.Get("name").(string)
 		}
 		if d.HasChange("description") {
-			updateOpts.Description = d.Get("description").(string)
+			desc := d.Get("description").(string)
+			updateOpts.Description = &desc
```
Comment (Lance52259) [inline huaweicloud/services/elb/resource_huaweicloud_elb_listener.go:324]: Using function `golangsdk.MaybeString()` can save the memory space of a variable.

Comment (Jason-Zhang9309) [reply huaweicloud/services/elb/resource_huaweicloud_elb_listener.go:324]: This function returns nil when the string is empty, that is not the original purpose of these lines of code.

Comment (ShiChangkuo) [reply huaweicloud/services/elb/resource_huaweicloud_elb_listener.go:324]: when description is empty, MaybeString will return nil which will be omitted in request body.

**Pull request merged**